### ISSUE TITLE
Get-Content with -AsByteStream -ReadCount X -Raw should return byte arrays

### ISSUE THREAD

**Issue by rhubarb-geek-nz** [PowerShell/PowerShell#21510]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Get-Content with -AsByteStream, -Raw and -ReadCount should return list of byte []

Simple script below

```
Set-Content -LiteralPath 'foo.bar' -Value 'foobar' -NoNewLine

Get-ChildItem -LiteralPath 'foo.bar'

$foo = Get-Content -LiteralPath 'foo.bar' -AsByteStream

$foo.GetType().Name
$foo[0].GetType().Name
$foo.Count
$foo[0].Count

$foo = Get-Content -LiteralPath 'foo.bar' -AsByteStream -ReadCount 3

$foo.GetType().Name
$foo[0].GetType().Name
$foo.Count
$foo[0].Count
$foo[1].Count

$foo = Get-Content -LiteralPath 'foo.bar' -AsByteStream -ReadCount 3 -Raw

$foo.GetType().Name
$foo[0].GetType().Name
$foo.Count
$foo[0].Count
$foo[1].Count
```

### Expected behavior

```console
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---          21/04/2024    03:03              6 foo.bar
Object[]
Byte
6
1
Object[]
Object[]
2
3
3
Object[]
Byte[]
2
3
3
```

### Actual behavior

```console
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---          21/04/2024    03:03              6 foo.bar
Object[]
Byte
6
1
Object[]
Object[]
2
3
3
Object[]
Byte[]
1
6
0
```

### Error details

```console
no errors
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.2
PSEdition                      Core
GitCommitId                    7.4.2
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

just the above text will suffice

**Comment by rhubarb-geek-nz** [comment]
Compare with proof of concept [Spit-Content](https://www.powershellgallery.com/packages/rhubarb-geek-nz.SplitContent/1.0.0)

```
$foo = Split-Content -LiteralPath 'foo.bar' -AsByteStream -ReadCount 3

$foo.GetType().Name
$foo[0].GetType().Name
$foo.Count
$foo[0].Count
$foo[1].Count
```

Gives list of byte arrays

```
Object[]
Byte[]
2
3
3
```

**Comment by mklement0** [comment]
I support this feature request, but let me add some context:

*  If `-ReadCount` is specified, `-Raw` is _ignored_ (which is unfortunate).

* `-Raw` in combination with `-AsByteStream` - but without `-ReadCount` -  _does_ emit a `[byte[]]` array, but it invariably means reading the _whole_ input file into memory.

* `-ReadCount` - wich currently emits `[object[]]` arrays - should:
  * emit `[string[]]` arrays by default 
  * if `-AsByteStream` is also specified, should emit `[byte[]]` arrays

In other words: when "chunking" is requested via `-ReadCount` a _strongly typed_ array should be returned, which helps performance, especially in v7.4+ with `-AsByeStream`, which then allows directly outputting chunked byte data to an _external (native) program_.

**Comment by jdhitsolutions** [comment]
Added for the Cmdlet WG to review

**Comment by JamesWTruher** [comment]
The WG discussed this and while we recognize this behavior may not be consistent with expectations, we believe that this does not meet the bar for return on engineering investment to make any changes.

**Comment by mklement0** [comment]
> does not meet the bar for return on engineering investment 

Isn't the proper resolution then to assign the `Up-for-grabs` label?
